Building container images from `scratch` doesn't work

### Describe the bug

This section of the documentation is slightly outdated:
https://quarkus.io/guides/building-native-image#build-a-container-image-from-scratch

Zlib is not available anymore at the URL:
```
https://www.zlib.net/zlib-1.2.13.tar.gz
```
should instead be:
```
https://www.zlib.net/zlib-1.3.tar.gz
```

Additionally, I have been unable to compile using the proposed base image:
```
quay.io/quarkus/ubi-quarkus-graalvmce-builder-image:jdk-17
```
Fails with:
```
#0 29.63 [INFO] ------------------------------------------------------------------------
#0 29.63 [INFO] BUILD FAILURE
#0 29.63 [INFO] ------------------------------------------------------------------------
#0 29.64 [INFO] Total time:  26.804 s
#0 29.64 [INFO] Finished at: 2023-10-20T14:09:54Z
#0 29.64 [INFO] ------------------------------------------------------------------------
#0 29.64 [ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:2.4.2.Final:build (default) on project getting-started: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
#0 29.64 [ERROR]        [error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: java.lang.IllegalStateException: Out of date version of GraalVM detected: native-image 17.0.8 2023-07-18. Quarkus currently supports 21.2. Please upgrade GraalVM to this version.
#0 29.64 [ERROR]        at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.checkGraalVMVersion(NativeImageBuildStep.java:378)
#0 29.64 [ERROR]        at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:171)
#0 29.64 [ERROR]        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
#0 29.65 [ERROR]        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
#0 29.65 [ERROR]        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
#0 29.65 [ERROR]        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
#0 29.65 [ERROR]        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:821)
#0 29.65 [ERROR]        at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
#0 29.65 [ERROR]        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
#0 29.65 [ERROR]        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
#0 29.65 [ERROR]        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
#0 29.65 [ERROR]        at java.base/java.lang.Thread.run(Thread.java:833)
#0 29.65 [ERROR]        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
#0 29.65 [ERROR] -> [Help 1]
```

Using as base image:
```
quay.io/quarkus/ubi-quarkus-graalvmce-builder-image:22.1-java17
```
I get a different error:
```
0 31.75 Error: Could not find target method: public static void io.quarkus.runtime.graal.Target_sun_java2d_cmm_lcms_LCMS_JDK17.colorConvert(io.quarkus.runtime.graal.Target_sun_java2d_cmm_lcms_LCMS_JDK17$LCMSTransform,io.quarkus.runtime.graal.Target_sun_java2d_cmm_lcms_LCMS_JDK17$LCMSImageLayout,io.quarkus.runtime.graal.Target_sun_java2d_cmm_lcms_LCMS_JDK17$LCMSImageLayout)
#0 31.75 com.oracle.svm.core.util.UserError$UserException: Could not find target method: public static void io.quarkus.runtime.graal.Target_sun_java2d_cmm_lcms_LCMS_JDK17.colorConvert(io.quarkus.runtime.graal.Target_sun_java2d_cmm_lcms_LCMS_JDK17$LCMSTransform,io.quarkus.runtime.graal.Target_sun_java2d_cmm_lcms_LCMS_JDK17$LCMSImageLayout,io.quarkus.runtime.graal.Target_sun_java2d_cmm_lcms_LCMS_JDK17$LCMSImageLayout)
#0 31.75        at com.oracle.svm.core.util.UserError.abort(UserError.java:72)
#0 31.75        at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.findOriginalMethod(AnnotationSubstitutionProcessor.java:764)
#0 31.75        at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.handleMethodInAliasClass(AnnotationSubstitutionProcessor.java:381)
#0 31.75        at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.handleAliasClass(AnnotationSubstitutionProcessor.java:353)
#0 31.75        at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.handleClass(AnnotationSubstitutionProcessor.java:323)
#0 31.75        at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.init(AnnotationSubstitutionProcessor.java:279)
#0 31.75        at com.oracle.svm.hosted.NativeImageGenerator.createAnnotationSubstitutionProcessor(NativeImageGenerator.java:933)
#0 31.75        at com.oracle.svm.hosted.NativeImageGenerator.setupNativeImage(NativeImageGenerator.java:846)
#0 31.75        at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:555)
#0 31.75        at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:515)
#0 31.75        at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:407)
#0 31.75        at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:585)
#0 31.75        at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:128)
#0 31.75        at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:615)
```

and the only working base image I found is:
```
quay.io/quarkus/ubi-quarkus-graalvmce-builder-image:22.1-java11
```

but I think that it should work on Java 17 too.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Mandrel or GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

cc. @cescoffier

--- a/docs/src/main/asciidoc/building-native-image.adoc
+++ b/docs/src/main/asciidoc/building-native-image.adoc
@@ -705,7 +705,7 @@ COPY --chown=quarkus:quarkus pom.xml /code/
 RUN mkdir /musl && \
     curl -L -o musl.tar.gz https://more.musl.cc/11.2.1/x86_64-linux-musl/x86_64-linux-musl-native.tgz && \
     tar -xvzf musl.tar.gz -C /musl --strip-components 1 && \
-    curl -L -o zlib.tar.gz https://www.zlib.net/zlib-1.2.13.tar.gz && \
+    curl -L -o zlib.tar.gz https://www.zlib.net/zlib-1.3.tar.gz && \
     mkdir zlib && tar -xvzf zlib.tar.gz -C zlib --strip-components 1 && \
     cd zlib && ./configure --static --prefix=/musl && \
     make && make install && \

--- a/docs/src/main/asciidoc/building-native-image.adoc
+++ b/docs/src/main/asciidoc/building-native-image.adoc
@@ -705,7 +705,7 @@ COPY --chown=quarkus:quarkus pom.xml /code/
 RUN mkdir /musl && \
     curl -L -o musl.tar.gz https://more.musl.cc/11.2.1/x86_64-linux-musl/x86_64-linux-musl-native.tgz && \
     tar -xvzf musl.tar.gz -C /musl --strip-components 1 && \
-    curl -L -o zlib.tar.gz https://www.zlib.net/zlib-1.3.tar.gz && \
+    curl -L -o zlib.tar.gz https://github.com/madler/zlib/releases/download/v1.2.13/zlib-1.2.13.tar.gz && \
     mkdir zlib && tar -xvzf zlib.tar.gz -C zlib --strip-components 1 && \
     cd zlib && ./configure --static --prefix=/musl && \
     make && make install && \

